source/module_elecstate/elecstate_pw.cpp

```diff
@@ -528,4 +528,59 @@ template class ElecStatePW<std::complex<float>, base_device::DEVICE_GPU>;
 template class ElecStatePW<std::complex<double>, base_device::DEVICE_GPU>;
 #endif 
 
+template<typename T, typename Device>
+void ElecStatePW<T, Device>::cal_tau(const psi::Psi<T, Device>& psi)
```

Comment:
maybe it will be better to put this function into another file?

Reply:
I have move it to a new file.